server/data/mappers/timelineMapper.ts

```diff
@@ -1,13 +1,23 @@
+import { parseISO } from 'date-fns'
```

Comment:
The `timelineMapper` needed some love ...

The purpose of the `timelineMapper` is to map a `TimelineResponse` which comes from the API into a `Timeline` which is a view model object.
For the most part they are field by field identical, with the exception of the event timestamp which is an ISO formatted string in the API response, and is a `Date` in the view model.
That was all fine ...

However, the `TimelineResponse` from the API includes ALL timeline event types. The API will record timeline events for lots of different events, but the UI only supports a subset of those. Therefore to correctly map from the API events to the view model events we need to filter the event types to just those that are supported.
Previously we did this in the service layer as technically it's too late (ie. how can this mapper return a valid view model instance for a given event type if the view model doesnt support that event type)